Update Server Docs (#138)
vircadia/vircadia-docs-sphinx: docs/source/host/start-hosting.rst

Vircadia's hosting interface consists of the following components that work together to create an immersive VR experience for your visitors:

* The *Domain Server* is a virtual world that gives you a space to build content in.

> * A *domain* is a registered server in the metaverse. Each domain is assigned a unique ID to identify it. You can connect your domain to the metaverse by logging into your account via the Vircadia website.

Reviewer: ```suggestion
* A *domain* is a server in the metaverse. Each domain is assigned a unique ID to identify it. You can connect your domain to a Metaverse Server.
```

Author: I think that the specific reference to the Vircadia website will make the most sense until we have federation working and UX can easily make obvious how it all connects.

Reviewer: ```suggestion
* A *domain* is a server in the metaverse. Each domain is assigned a unique ID to identify it. You can connect your domain to the Vircadia Metaverse Server by logging into your account on the Vircadia website.
```
Yeah I guess so. How about this then?
